File: Admin_p/Admin_p.Web/Modules/ServerTypes/Electro/ProductsForm.ts
import { StringEditor, DecimalEditor, IntegerEditor, ServiceLookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProductsForm {
    Name: StringEditor;
    Description: StringEditor;
    Price: DecimalEditor;
    StockQuantity: IntegerEditor;
    CategoryId: ServiceLookupEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
    ImagePath: StringEditor;
}

export class ProductsForm extends PrefixedContext {
    static readonly formKey = 'Electro.Products';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductsForm.init)  {
            ProductsForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = IntegerEditor;
            var w3 = ServiceLookupEditor;
            var w4 = DateEditor;

            initFormType(ProductsForm, [
                'Name', w0,
                'Description', w0,
                'Price', w1,
                'StockQuantity', w2,
                'CategoryId', w3,
                'CreatedAt', w4,
                'UpdatedAt', w4,
                'ImagePath', w0
            ]);
        }
    }
}

File: Admin_p/Admin_p.Web/Modules/Electro/Products/ProductsForm.cs
using Serenity.ComponentModel;
using System;

namespace Admin_p.Electro.Forms;

[FormScript("Electro.Products")]
[BasedOnRow(typeof(ProductsRow), CheckNames = true)]
public class ProductsForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public int CategoryId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string ImagePath { get; set; }
}